week05/clause_test.go

```diff
@@ -0,0 +1,82 @@
+package demo
+
+import (
+	"testing"
+)
+
+func emptyClauses(t testing.TB) *Clauses {
```

Comment:
The `Clauses` type is a `map[string]interface{}`. 

A map, under the covers is already a pointer, you almost never need the pointer to a map.

Please update your code to just you `Clauses` and not `*Clauses`.